### PR TITLE
De-globalize, simplify hooks.

### PR DESCRIPTION
**Pull request by Road-block**
Following a more modular approach.
- Wrap addon code in a table so we don't pollute the global environment.
- Better Hook implementation: Only replace the functions we need, always call the original if we don't handle things to play nice with other addon hooks.
- Add a Print function for debug output.

### CODE REVIEW

File: MMC.lua [hunk 1]
```diff
@@ -180,25 +188,14 @@ function MMC_VerifyIdentity(target)
...
-        if not msg or not fun then
-            return false;
-        else
-            fun(msg);
-            
-            if editBox then
-                editBox:AddHistoryLine(text);
-            end
-            return false;
-        end
```
Comment (DennisWG) [inline MMC.lua]: Removing this breaks things like 
```lua
/cast [stance:1] Enrage; Dire Bear Form
```

Dire bear Form will never get cast here. We sill need something like
```lua
    if not conditionals then
        if not msg then
            return false;
        else
            MMC.Hooks.CAST_SlashCmd(msg);
            return true;
        end
    end
```

When you add this, I will gladly approve this wonderful pull request!

Comment (Road-block) [inline MMC.lua]: Sorry for the late reply,
Comment coincided with my messed up sleep cycle.

I'm glad you figured it out 👍  and sorry for the bug

**Comment by Road-block** [comment]
Your fix should be coming in with the latest commit, sorry for the oversight (don't know how nested comments are handled in notifications, so just in case I post it here too)

File: MMC.lua (hunk 1 above)
Comment (DennisWG) [inline MMC.lua]: No worries. After all, you're helping me here big time!

**Pull request merged**